syntax ="proto3";

/*
Import timestamp in proto file of protobuf for GRPC
developers.google.com/protocol-buffers/docs/reference/go-generated#well-known-types
*/
import 'google/protobuf/timestamp.proto';


/* 
from db folder the init-db.sql  & models.py
*/
message Person {
	int32 id=1;
	string first_name = 2;
	string last_name = 3;
	string company_name = 4;
}

message Location {
	int32 id = 1;
	int32 person_id = 2;
	string longitude = 3;
	string latitude = 4;
	google.protobuf.Timestamp creation_time = 5;
	string wkt_shape = 6;
}

message ConnectionMesg {
	Person person = 1;
	Location location = 2;
}

message ConnectionMesgList {
	repeated ConnectionMesg connections = 1;
}

/*
from localhost:30001
 */ 
message SearchMesg {
	int32 person_id = 1;
	google.protobuf.Timestamp start_date = 2;
	google.protobuf.Timestamp end_date = 3;
	float meters =4;	
}

service ConnectionService {
	rpc FindContacts(SearchMesg) returns ( ConnectionMesgList);
}